test/lib/router.js

```diff
@@ -849,35 +849,6 @@ describe('Router', function () {
       expect(router.stack[0]).to.have.property('path', '/one');
       expect(router.stack[1]).to.have.property('path', '/two');
     });
-
-    it('resolves non-parameterized routes without attached parameters', function(done) {
-      var app = new Koa();
-      var router = new Router();
-
-      router.get('/notparameter', function (ctx, next) {
-        ctx.body = {
-          param: ctx.params.parameter,
-        };
-      });
-
-      router.get('/:parameter', function (ctx, next) {
-        ctx.body = {
-          param: ctx.params.parameter,
-        };
-      });
-
-      app.use(router.routes());
-      request(http.createServer(app.callback()))
-        .get('/notparameter')
-        .expect(200)
-        .end(function (err, res) {
-          if (err) return done(err);
-
-          expect(res.body.param).to.equal(undefined);
-          done();
-        });
-    });
```

Comment:
@smkoyan this test is important and removing it will not allow us to check use cases where users want a specific route to be hit followed by a catch-all route.

```js
router.get('/specific-route') // Hit when specific-route is provided
router.get('/:catchall') // Hit in any other case with a capture
```

Reply:
Addressed in https://github.com/ZijianHe/koa-router/pull/508